make/mt-os/mt-esp32.mk

```diff
@@ -102,6 +102,14 @@ endif
 # mt-tls-mbedtls-esp32.mk. Or delete the line below to turn off TLS lib cross compilation.
 include make/mt-config/mt-tls-$(XI_BSP_TLS)-esp32.mk
 
+ifeq ($(XI_BSP_TLS),wolfssl)
+# WolfSSL TLS BSP configuration
+XI_CONFIG_FLAGS += -DXI_PROVIDE_WOLFSSL_SEED_GENERATOR
+XI_CONFIG_FLAGS += -DXI_PROVIDE_WOLFSSL_XTIME_XGMTIME
+else ifeq ($(XI_BSP_TLS),mbedtls)
+XI_TLS_LIB_INC_DIR = $(IDF_PATH)/components/mbedtls/include
```

Comment:
so this means the default is to use the SDK's mbedTLS version. Is this ok?

Reply:
The PRESET's default is still to use WolfSSL, but maybe I should change it. mbedTLS does have hardware acceleration in the ESP32, while wolf doesn't....

What do you think?

Reply:
what if I set the `XI_USE_EXTERNAL_TLS_LIB=0` in the make command line? What will happen? Which mbedTLS gets used? I think this line will be executed still which means the SDK's mbedTLS lib will be used. Won't it?

Reply:
So I see two value assignment to `XI_TLS_LIB_INC_DIR`. One in `mt-tls-mbedtls.mk` and one here `mt-esp32.mk`. In case `XI_USE_EXTERNAL_TLS_LIB == 0` both get executed. First the one in `mt-tls-mbedtls.mk` then this here in `mt-esp32.mk`, the second overrides the first. Which means setting NOT to use external TLS lib still assigns the external SDK's path.
Isn't this so?
But the final question is if the `XI_USE_EXTERNAL_TLS_LIB` flag works for both cases `1` and `0`?

Reply:
Oh,  nice catch! 👍  You're right, that's broken. I can easily fix it with another `ifeq($(XI_USE_EXTERNAL_TLS_LIB),1)`, and I will.

`XI_USE_EXTERNAL_TLS_LIB=0` will still NOT work for the ESP32 either way, though. The example would still link the SDK's mbedTLS by default, so that would have to be solved, and things start getting more convoluted.

I can't think of a good reason to use our non-hardware-accelerated version instead of the SDK's, which is why `XI_USE_EXTERNAL_TLS_LIB` wasn't originally overridable from the command line.